### PR TITLE
ci: add pre-commit to lint for general file style

### LINKED ISSUE #111: ci: install precommit to lint all files
We have extensive style support for all of the languages inside envoy-mobile.

In addition we should add https://pre-commit.com/ to the repo to check against common file style like newlines, and trailing spaces.

### PR DESCRIPTION
**Pull request by junr03**
Signed-off-by: Jose Nino <jnino@lyft.com>

Description: this PR introduces pre-commit in order to lint for general file style. All of our linters are language specific. This linter adds common style, e.g newline at the end of files.
Risk Level: low
Testing: ran locally and added CI job.
Docs Changes: added docs in STYLE.md

Fixes #111

### CODE REVIEW

File: STYLE.md [hunk 1]
```diff
@@ -1,3 +1,13 @@
+## General file style
+
+- General style guidelines -- like newlines at the end of file -- are linted
+for using [pre-commit](https://pre-commit.com/)
+- The rules enfornced are available in the
```
Comment (rebello95) [inline STYLE.md]:
```suggestion
- The rules enforced are available in the
```

Comment (junr03) [reply STYLE.md]: hmmm spell checker from all_lint should catch that. Let me take a look at that.

**Comment by junr03** [comment]
Output from CI:

```
##[section]Starting: Running pre-commit
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.151.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents:
pre-commit run --all-files
========================== Starting Command Output ===========================
[command]/bin/bash --noprofile --norc /Users/vsts/agent/2.153.2/work/_temp/1a88a8db-0b09-4848-8d48-f8847ea9d614.sh
[INFO] Initializing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Trim Trailing Whitespace.................................................Passed
Fix End of Files.........................................................Passed
Check docstring is first.................................................Passed
Check that executables have shebangs.....................................Passed
Check for merge conflicts................................................Passed
Check Yaml...............................................................Passed
Detect Private Key.......................................................Passed
Mixed line ending........................................................Passed
##[section]Finishing: Running pre-commit
```

File: STYLE.md (hunk 1 above)
Comment (junr03) [reply STYLE.md]: nvm we don't have that on. Will add that.

**Pull request merged**